collector/lib/CollectorConfig.h

```diff
@@ -93,6 +108,14 @@ class CollectorConfig {
 
   static std::pair<option::ArgStatus, std::string> CheckConfiguration(const char* config, Json::Value* root);
 
+  void SetRuntimeConfig(sensor::CollectorConfig runtime_config) {
+    runtime_config_ = std::move(runtime_config);
+  }
```

Comment:
I know we discussed this offline yesterday, but if we don't expect the caller to keep using the `runtime_config` object, we should probably just use an rvalue here:
```suggestion
  void SetRuntimeConfig(sensor::CollectorConfig&& runtime_config) {
    runtime_config_ = runtime_config;
  }
```
Alternatively, method overloading is a thing, we can have both and let the caller decide if they are `std::move`ing the object or not.